main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -2156,4 +2155,15 @@ object ParsedAst {
     * Represents an optional purity and optional effect.
     */
   case class PurityAndEffect(pur: Option[Type], eff: Option[EffectSet])
+
+  /**
+    * A foreach expression generator, i.e. `x <- xs`.
+    *
+    * @param sp1 the position of the first character in the expression.
+    * @param pat the pattern on the left hand side.
+    * @param exp the iterable expression.
+    * @param sp2 the position of the last character in the expression.
+    */
+  case class ForEachGenerator(sp1: SourcePosition, pat: ParsedAst.Pattern, exp: ParsedAst.Expression, sp2: SourcePosition)
```

Comment:
We can use this name, until we come up with a better one.

I guess we will have more of these when we add the other for loop.